packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Popover from "@paprika/popover";
+import Confirmation from "./components/Confirmation";
+import Divider from "./components/Divider";
+import Trigger from "./components/Trigger";
+import LinkItem from "./components/LinkItem";
+import Item from "./components/Item";
+import contentStyles from "./DropdownMenuContentStyles.styles";
+
+const { alignTypes, oneOf, node, func } = PropTypes;
+
+const propTypes = {
+  /** Alignment of the dropdown menu */
+  align: oneOf(alignTypes),
+
+  /** Children should consist of <Dropdown.Item /> */
+  children: node.isRequired,
+
+  /** Render prop for rendering the trigger element that toggles the dropdown */
+  renderTrigger: func.isRequired,
+};
+
+const defaultProps = {
+  align: "bottom",
+};
+
+const DropdownMenu = props => {
+  const { align, children, ...moreProps } = props;
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isConfirming, setIsConfirming] = React.useState(false);
+  const [renderConfirmation, setRenderConfirmation] = React.useState(null);
+  const triggerRef = React.useRef(null);
+
+  const handleCloseMenu = () => {
+    setIsOpen(false);
+    setIsConfirming(false);
+    setRenderConfirmation(null);
+    if (triggerRef.current) triggerRef.current.focus();
+  };
+
+  const handleOpenMenu = () => {
+    setIsOpen(true);
+  };
+
+  const getTriggerStateAndHelpers = () => ({
+    isOpen,
+    handleOpenMenu,
+  });
+
+  const handleShowConfirmation = renderConfirmation => () => {
+    setIsConfirming(prevIsConfirmingState => !prevIsConfirmingState);
+    setRenderConfirmation(prevIsConfirmingState => (prevIsConfirmingState ? null : renderConfirmation));
+  };
+
+  const randomIdString = Math.random()
```

Comment:
This will change on every render—probably not something we want to do. We can store this value in a ref